src/configuration.ts

```diff
@@ -207,13 +217,22 @@ export function loadConfigurationFromPath(configFilePath?: string, originalFileP
         return DEFAULT_CONFIG;
     } else {
         const resolvedConfigFilePath = resolveConfigurationPath(configFilePath);
+        const resolvedConfigFileExt = path.extname(resolvedConfigFilePath);
         let rawConfigFile: RawConfigFile;
-        if (path.extname(resolvedConfigFilePath) === ".json") {
-            const fileContent = stripComments(fs.readFileSync(resolvedConfigFilePath)
-                .toString()
-                .replace(/^\uFEFF/, ""));
+        if (/\.(json|ya?ml)/.test(resolvedConfigFileExt)) {
+            const fileContent = fs.readFileSync(resolvedConfigFilePath, "utf8").replace(/^\uFEFF/, "");
             try {
-                rawConfigFile = JSON.parse(fileContent) as RawConfigFile;
+                if (resolvedConfigFileExt === ".json") {
+                    rawConfigFile = JSON.parse(stripComments(fileContent)) as RawConfigFile;
+                } else { // /\.ya?ml/.test(resolvedConfigFileExt) === true
```

Comment:
what's this comment doing here? if it's stray code, please delete. if it's explanatory, write some additional words at the beginning (don't start with "/") and put it on its own line